### ISSUE TITLE
upgrade to 1.12.0 breaks build

### ISSUE THREAD

**Issue by mazdack** [allegro/axion-release-plugin#348]
going from 1.11.0 to 1.12.0 and getting error on building project
project is kotlin, gradle + gradle dsl

error:
```
class org.codehaus.groovy.runtime.GStringImpl cannot be cast to class java.lang.String (org.codehaus.groovy.runtime.GStringImpl is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @46d56d67; java.lang.String is in module java.base of loader 'bootstrap')
```
config is:
```
scmVersion {
    useHighestVersion = true
    versionCreator("versionWithBranch")
    versionIncrementer("incrementMinor")

    checks(closureOf<ChecksConfig> {
        uncommittedChanges = false
        aheadOfRemote = false
        snapshotDependencies = true
    })
}
```

**Comment by SteKoe** [comment]
I figured out how to fix this issue and provided a PR. All tests still pass. GStringImpl is now dropped and java.lang.String is returned instead. C.f. #351

**Comment by adamdubiel** [comment]
Thanks for providing solution! I merged it, there is one fix waiting, so i should release 1.12.1. Sorry for slow response time, but for some time i've been swamped with other work and only able to do work in here in my spare time.

**Comment by SteKoe** [comment]
No worries, thanks for providing this library for free! Your work is much appreciated :) keep on!
